### PR TITLE
Signature Component

### PR DESCRIPTION
**Pull request by evancloutier**
Introduces the signature component into `ws-directives`, which uses a dependency `signature_pad` to create signatures using a canvas.

signature pad

When 'Save signature' is pressed, the binding function `onSave` is called to propagate that PNG data up to the parent controller for use. When 'Redo' is pressed, that PNG is cleared from the parent controller attribute.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -40,7 +40,7 @@ Check `peerDependencies` in `package.json` for specific versions.
 ## local development
 
 1. run `npm install` to install dev dependencies
-2. run `npm start:dev` o watch & build files, and serve up the examples page.
+2. run `npm run start:dev` o watch & build files, and serve up the examples page.
```
Comment (SethDavenport) [inline README.md]: thanks for fixing this.

File: src/signature-pad/ws-signature-pad.component.js [hunk 2]
```diff
@@ -0,0 +1,12 @@
+import template from './ws-signature-pad.html';
+
+const wsSignaturePadComponent = {
+  template,
+  controller: 'wsSignaturePadController',
```
Comment (SethDavenport) [inline src/signature-pad/ws-signature-pad.component.js]: Under the new model, you can _probably_ do:

```js
import wsSignaturePadController from './ws-signature-pad.controller';

const wsSignaturePadComponent {
  template,
  controller: wsSignaturePadContoller, // function reference, not a literal
  // ...
}
```

Then you don't have to remember to attach the controller to `angular.module`.

But it's also fine to leave it like this.

Comment (evancloutier) [inline src/signature-pad/ws-signature-pad.component.js]: Agree w/ this, will update 💯

File: src/signature-pad/ws-signature-pad.controller.js [hunk 3]
```diff
@@ -0,0 +1,71 @@
+import SignaturePad from 'signature_pad';
+
+wsSignaturePadController.$inject = [
+  '$scope',
+  '$window',
+  '$element',
+];
+
+export default function wsSignaturePadController($scope, $window, $element) {
```
Comment (SethDavenport) [inline src/signature-pad/ws-signature-pad.controller.js]: This is also fine as-is. But out of curiosity how do you feel about class-based controllers? E.g.

```js
export default class WsSignaturePadController {
  static $inject = ['$scope', '$window', '$element'];
  constructor($scope, $window, $element) {
    this.$scope = $scope;
    this.$window = $window;
    this.$element = $element;
  }

  $ngOnInit = () => {
    this.state = {
      isEdited: false,
      isSaved: false,
      signatureData: null,
    }
  };

  clear = () => {
    this.signaturePad.clear();
    this.state.isEdited = false;
  }

  // etc.
}
```

File: src/signature-pad/ws-signature-pad.controller.js [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+    _resizeCanvas();
+  }
+
+  function _resizeCanvas() {
+    canvas.width = canvasContainer.offsetWidth;
+    canvas.height = canvasContainer.offsetHeight;
+  }
+
+  function isEdited() {
+    $scope.state.isEdited = true;
```
Comment (SethDavenport) [inline src/signature-pad/ws-signature-pad.controller.js]: I'm surprised to see this living on `$scope`: I thought `component` bound the view to the controller instance directly?

**Comment by SethDavenport** [comment]
Some 'food for thought' type comments but the implementation looks good. :shipit:

File: src/signature-pad/_ws-signature-pad.scss [hunk 5]
```diff
@@ -0,0 +1,46 @@
+ws-signature-pad {
+  .signature-pad-container {
+    position: relative;
+    min-width: 200px;
+    background-color: #ffffff;
```
Comment (DanGodfrey) [inline src/signature-pad/_ws-signature-pad.scss]: let's used `$color-white` here

File: src/signature-pad/ws-signature-pad.html [hunk 6]
```diff
@@ -0,0 +1,20 @@
+<div class="signature-pad-container">
+  <div class="signature-pad-body">
+    <canvas></canvas>
+  </div>
+  <div class="row-xs signature-pad-footer">
+    <div class="col-6 clear">
+      <button ng-click="clear()"
+        ng-if="!state.isSaved && state.isEdited">Clear</button>
```
Comment (DanGodfrey) [inline src/signature-pad/ws-signature-pad.html]: strings will need to be translated.

Comment (SethDavenport) [inline src/signature-pad/ws-signature-pad.html]: Possibly a topic for another time - does it make sense to maintain a string table inside this library or to ask consumers to pass their text bits in as attributes?

Comment (DanGodfrey) [inline src/signature-pad/ws-signature-pad.html]: @SethDavenport thats a good point. We currently do a mix of this but should really make it consistent.

Comment (SethDavenport) [inline src/signature-pad/ws-signature-pad.html]: One of the main consumers of ws-directives built-in string table seems to be ws-footer: a component that doesn't really belong here and isn't being used anyway: https://trello.com/c/utZ4qvXh/16-ws-directives-is-it-really-the-right-place-for-footerdirective

**Pull request merged**